### PR TITLE
ENG-120: update wallet connect flow

### PR DESCRIPTION
**Pull request by stormcloud266**

### CODE REVIEW

**Review by chair28980 — CHANGES_REQUESTED** [review]
@stormcloud266 Looking great! Can you also update the text above the button to read:

```
Connect your web3 wallet and sign the message to be part of an early player leaderboard.
```

image

**Comment by stormcloud266** [comment]
@chair28980 made the requested update and also changed the button text to "sign message" instead of just "sign"

**Review by chair28980 — APPROVED** [review]

File: components/wallet-email.tsx [hunk 1]
```diff
@@ -28,7 +28,14 @@ const WalletEmail = ({ puzzleId, successMessage }: ComponentProps) => {
     formState: { errors, isValid, isSubmitSuccessful },
   } = useForm<FormProps>();
 
+  const defaultClasses = "font-normal text-neutral-500";
+  const activeClasses = "font-bold text-neutral-50";
+  const completedClasses = defaultClasses + " line-through";
```
Comment (illepic) [inline components/wallet-email.tsx]: Let's use `clsx` for this: https://www.npmjs.com/package/clsx. Example: https://github.com/infinity-keys/ik-landing/blob/92b221db1a630d877ecfe9dc947c5b64366c53bf/components/puzzle.tsx#L111

File: components/wallet.tsx [hunk 2]
```diff
@@ -1,71 +1,75 @@
...
+      const status =
+        !userAccount && !userSignature
+          ? "connect"
+          : userAccount && !userSignature
+          ? "sign"
+          : "disconnect";
+
+      setStatus(status);
+    }
+  }, [userAccount, userSignature, setStatus]);
```
Comment (illepic) [inline components/wallet.tsx]: Let's talk about `useEffect` in this context.

**Review by illepic — CHANGES_REQUESTED** [review]

File: components/wallet.tsx [hunk 3]
```diff
@@ -1,72 +1,70 @@
...
-      disconnect();
-    }
-
-  }
-
-  const disconnect = () => {
-    wallet.clear();
+const Wallet = ({ onWalletSignature }: WalletProps) => {
+  const { current, send, isSigned, isLoading, isNotConnected } =
+    useWalletConnect();
```
Comment (illepic) [inline components/wallet.tsx:11]: Hell yeah.

File: components/wallet.tsx [hunk 4]
```diff
@@ -1,72 +1,70 @@
...
-      {!account && !signature && 'Connect wallet'}
-      {account && !signature && 'Waiting on signature'}
-      {account && signature && 'Disconnect'}
-    </button>
+    <>
+      <p
+        className={clsx(
+          "text-md mb-3",
+          isNotConnected || current.matches("connecting")
+            ? "font-bold text-neutral-50"
```
Comment (illepic) [inline components/wallet.tsx:23]: Did you mean `50` or `500` here?

Comment (illepic) [reply components/wallet.tsx:23]: And if you meant `500`, then move `text-neutral-500` to the non-logic styles.

File: tsconfig.json [hunk 5]
```diff
@@ -17,7 +17,8 @@
     "incremental": true,
     "paths": {
       "@lib/*": ["lib/*"],
-      "@components/*": ["components/*"]
+      "@components/*": ["components/*"],
+      "@hooks/*": ["hooks/*"]
```
Comment (illepic) [inline tsconfig.json:21]: 💥

**Review by illepic — CHANGES_REQUESTED** [review]
Just a quick clarification.

File: components/wallet.tsx (hunk 4 above)
Comment (stormcloud266) [reply components/wallet.tsx:23]: I meant `50` just so it's clear what step the user is on

Screen Shot 2022-07-11 at 8 38 34 AM

**Review by illepic — APPROVED** [review]

**Pull request merged**